### DISCUSSION TITLE
Is it possible to also detect Firewire detachments?

### DISCUSSION THREAD

**Discussion by michaellammers** [himbeles/mac-device-connect-daemon#8] [Q&A]
I have a sound card (Focusrite Saffire Pro 40 Firewire Audio Interface) which is critical for a show and it would be great if we can get an alert if the card is not detected anymore.

**Comment by himbeles** [comment]
Hi Michael,

See 

https://github.com/himbeles/mac-device-connect-daemon?tab=readme-ov-file#setup-the-daemon

The device class for FireWire should be **IOPCI**

So, you try to query for the sound card with 

`ioreg -p IOPCI -l`

and check if you can detect it. 

Unfortunately, the daemon currently does not allow to detect disconnects. I'm working on adding FireWire support to my app Stecker. I'll keep you posted here as soon as I have something working.

**Reply by michaellammers** [reply]
Thanks! I need it for disconnects, but will first try detection somewhere this week and post the result here. Would be great if this eventually could be implemented in Stecker 👌

**Comment by michaellammers** [comment]
Don't know if it's useful, but I've added the output of the command `ioreg -p IOPCI -l` as an attachment here.
[ioreg_output.txt](https://github.com/user-attachments/files/17463434/ioreg_output.txt)
